### ISSUE TITLE
GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined

### ISSUE THREAD

**Issue by toshiro92** [OpenCTI-Platform/opencti#540]
Please replace every line in curly brackets { like this } with an appropriate answer, and remove this line.

## Description

With the latest version of OpenCTI (3.0), I'm facing the following error after all necessary containers were up:

error: [OPENCTI] GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined

## Environment

1. OS (where OpenCTI server runs): Docker image
2. OpenCTI version: OpenCTI 3.0
3. OpenCTI client: Frontend + Portainer for logs
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Run Grakn first and remove it from the original docker-compose.yml (To avoid failure on Grakn through docker-compose)
2. Execute the docker-compose.yml (Without default connectors)
3. Check every logs of all running containers:
- Redis
- Connector-OpenCTI
- Worker
- openCTI
- RabbitMQ
- Grakn
- Minio (Unhealthy)
4. OpenCTI complains with the following error:
```
$ node --max_old_space_size=8192 build/index.js
error: [OPENCTI] GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined
    at /opt/opencti/build/index.js:1:61829
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async ae (/opt/opencti/build/index.js:1:68247)
    at async de (/opt/opencti/build/index.js:1:69456)
    at async I (/opt/opencti/build/index.js:1:101527)
    at async /opt/opencti/build/index.js:1:123139
    at async n.b (/opt/opencti/build/index.js:1:123042)
    at async /opt/opencti/build/index.js:1:841022
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Expected Output

OpenCTI Up, running and accessible

## Actual Output

```
$ node --max_old_space_size=8192 build/index.js
error: [OPENCTI] GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined
    at /opt/opencti/build/index.js:1:61829
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async ae (/opt/opencti/build/index.js:1:68247)
    at async de (/opt/opencti/build/index.js:1:69456)
    at async I (/opt/opencti/build/index.js:1:101527)
    at async /opt/opencti/build/index.js:1:123139
    at async n.b (/opt/opencti/build/index.js:1:123042)
    at async /opt/opencti/build/index.js:1:841022
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
 
## Additional information

Not Yet

**Comment by SamuelHassine** [comment]
Hello @toshiro92,
 
Can you please look at the `build/index.js` and copy/paste the source code around the position 61829?

**Comment by oz17il** [comment]
Same issue. @SamuelHassine 
the file content and the file size are the same as in the source code. 

I tried to replace the file with a new one and it doesn't help.

OpenCTI version: 3.0.2
OS: Ubuntu 
After a new installation the same problem.

`error: [OPENCTI] GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined
    at /opt/opencti/build/index.js:1:62384
    at processTicksAndRejections (internal/process/task_queues.js:88:5)
    at async ae (/opt/opencti/build/index.js:1:68802)
    at async de (/opt/opencti/build/index.js:1:70011)
    at async I (/opt/opencti/build/index.js:1:102082)
    at async /opt/opencti/build/index.js:1:123694
    at async n.b (/opt/opencti/build/index.js:1:123597)
    at async /opt/opencti/build/index.js:1:847397
error Command failed with exit code 1.
`

**Comment by campeador** [comment]
Hi, 

I was getting this error too when I performed a grakn server clean. To fix this issue you should do 'yarn index' to create a new index.

opencti@opencti:/opencti$ /usr/local/bin/yarn serv
yarn run v1.22.0
$ node --max_old_space_size=8192 build/index.js
info: OpenCTI started in production mode with embedded file
info: [PRE-CHECK] > Grakn is alive
info: [PRE-CHECK] > Elasticsearch is alive
info: [PRE-CHECK] > Minio is alive
info: [PRE-CHECK] > Python3 is available
info: [INIT] > Grakn schema loaded
info: [INIT] > Elasticsearch indexes loaded
info: [MIGRATION] > Starting migration process
info: [MIGRATION] > Read 0 migrations from the database
info: [MIGRATION] > Migration process completed, platform is up to date
info: [INIT] > Platform default initialized
error: [OPENCTI] GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined
    at /home/opencti/docker_manual/opencti/build/index.js:1:62384
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async ae (/opencti/build/index.js:1:68802)
    at async de (l/opencti/build/index.js:1:70011)
    at async I (/opencti/build/index.js:1:102082)
    at async /opencti/build/index.js:1:123694
    at async n.b (/opencti/build/index.js:1:123597)
    at async /opencti/build/index.js:1:847397
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

opencti@opencti:/opencti$ /usr/local/bin/yarn index
yarn run v1.22.0
$ babel-node ./src/indexer.js
info: OpenCTI started in production mode with embedded file
info: > ---------------------------------------------------------------------
info: Indexing > Old indices deleted
info: Indexing > New indices created
info: Indexing done in 2.273 seconds (a few seconds)
Done in 5.89s.

opencti@opencti:/opencti$ /usr/local/bin/yarn serv
yarn run v1.22.0
$ node --max_old_space_size=8192 build/index.js
info: OpenCTI started in production mode with embedded file
info: [PRE-CHECK] > Grakn is alive
info: [PRE-CHECK] > Elasticsearch is alive
info: [PRE-CHECK] > Minio is alive
info: [PRE-CHECK] > Python3 is available
info: [INIT] > Grakn schema loaded
info: [INIT] > Elasticsearch indexes loaded
info: [MIGRATION] > Starting migration process
info: [MIGRATION] > Read 0 migrations from the database
info: [MIGRATION] > Migration process completed, platform is up to date
info: [INIT] > Initialization of settings and basic elements
info: [INIT] > Platform default initialized
info: [ADMIN_SETUP] admin user initialized
info: OPENCTI Ready on port 4000

**Comment by oz17il** [comment]
Hi @campeador 
Same problem...
root@ip-172-25-130-202:/opt/opencti# /usr/bin/yarn index
yarn run v1.22.0
$ babel-node ./src/indexer.js
info: 🚀 OpenCTI started in production mode with embedded file
info: > ---------------------------------------------------------------------
info: Indexing > Old indices deleted
info: Indexing > New indices created
info: Indexing of Settings done in 0.477 secs (a few seconds) - Element Avg: 0.48 secs
info: > ---------------------------------------------------------------------
info: Indexing of Role done in 0.573 secs (a few seconds) - Element Avg: 0.29 secs
info: > ---------------------------------------------------------------------
info: Indexing of Capability done in 2.119 secs (a few seconds) - Element Avg: 0.11 secs
info: > ---------------------------------------------------------------------
info: Indexing of Tag done in 0.525 secs (a few seconds) - Element Avg: 0.26 secs
info: > ---------------------------------------------------------------------
info: Indexing of Stix-Domain done in 1.247 secs (a few seconds) - Element Avg: 0.21 secs
info: > ---------------------------------------------------------------------
info: Indexing of Stix-Observable done in 0.61 secs (a few seconds) - Element Avg: 0.61 secs
info: > ---------------------------------------------------------------------
info: Indexing of relation_embedded done in 0.949 secs (a few seconds) - Element Avg: 0.24 secs
info: > ---------------------------------------------------------------------
info: Indexing of stix_relation_embedded done in 1.086 secs (a few seconds) - Element Avg: 0.22 secs
info: > ---------------------------------------------------------------------
info: Indexing done in 8.512 seconds (a few seconds)
Done in 10.99s.
root@ip-172-25-130-202:/opt/opencti# /usr/bin/yarn serv
yarn run v1.22.0
$ node --max_old_space_size=8192 build/index.js
info: 🚀 OpenCTI started in production mode with embedded file
info: [PRE-CHECK] > Grakn is alive
info: [PRE-CHECK] > Elasticsearch is alive
info: [PRE-CHECK] > Minio is alive
info: [PRE-CHECK] > Python3 is available
info: [INIT] > Grakn schema loaded
info: [INIT] > Elasticsearch indexes loaded
info: [MIGRATION] > Starting migration process
info: [MIGRATION] > Read 0 migrations from the database
info: [MIGRATION] > Migration process completed, platform is up to date
info: [INIT] > Platform default initialized
error: [OPENCTI] GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined
    at /opt/opencti/build/index.js:1:62384
    at processTicksAndRejections (internal/process/task_queues.js:88:5)
    at async ae (/opt/opencti/build/index.js:1:68802)
    at async de (/opt/opencti/build/index.js:1:70011)
    at async I (/opt/opencti/build/index.js:1:102082)
    at async /opt/opencti/build/index.js:1:123694
    at async n.b (/opt/opencti/build/index.js:1:123597)
    at async /opt/opencti/build/index.js:1:847397
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by SamuelHassine** [comment]
@oz17il, @toshiro92: please could you enable more logging in the configuration by using ` "logs_level": "debug"`?

**Comment by oz17il** [comment]
> @oz17il, @toshiro92: please could you enable more logging in the configuration by using ` "logs_level": "debug"`?

@SamuelHassine  Hi, what do you mean? can you provide the full command?

**Comment by toshiro92** [comment]
@SamuelHassine There it is:

```
yarn run v1.19.1
$ node --max_old_space_size=8192 build/index.js
info: 🚀 OpenCTI started in production mode with embedded file
info: [PRE-CHECK] > Grakn is alive
info: [PRE-CHECK] > Elasticsearch is alive
info: [PRE-CHECK] > Minio is alive
info: [PRE-CHECK] > Python3 is available
info: [INIT] > Grakn schema loaded
info: [INIT] > Elasticsearch indexes loaded
info: [MIGRATION] > Starting migration process
debug: [GRAKN - infer: undefined] Find > match $status isa MigrationStatus; get;
debug: [ELASTICSEARCH] loadByTerms > {"index":["stix_domain_entities_v2"],"_source_excludes":"rel_*","body":{"query":{"bool":{"must":[{"term":{"grakn_id.keyword":"V348200"}}]}}}}
debug: [GRAKN] Cache warning: V348200 should be available in cache
debug: [GRAKN - infer: false] getAttributes > MigrationStatus V348200
debug: [GRAKN - infer: undefined] Find > match $from isa MigrationStatus; $rel(status:$from, state:$to) isa migrate; get;
info: [MIGRATION] > Read 0 migrations from the database
info: [MIGRATION] > Migration process completed, platform is up to date
debug: [ELASTICSEARCH] paginate > {"index":"stix_domain_entities_v2","_source_excludes":"rel_*","track_total_hits":true,"body":{"from":0,"size":200,"sort":[],"query":{"bool":{"must":[{"match_all":{}},{"bool":{"should":[{"match_phrase":{"entity_type":"Settings"}},{"match_phrase":{"parent_types":"Settings"}}],"minimum_should_match":1}}],"must_not":[]}}}}
info: [INIT] > Platform default initialized
debug: [ELASTICSEARCH] loadByTerms > {"index":["stix_domain_entities_v2","stix_relations","stix_observables","work_jobs_index"],"_source_excludes":"rel_*","body":{"query":{"bool":{"must":[{"term":{"internal_id_key.keyword":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}},{"term":{"parent_types.keyword":"User"}}]}}}}
debug: [ELASTICSEARCH] loadByTerms > {"index":["stix_domain_entities_v2","stix_relations","stix_observables","work_jobs_index"],"_source_excludes":"rel_*","body":{"query":{"bool":{"must":[{"term":{"internal_id_key.keyword":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}},{"term":{"parent_types.keyword":"User"}}]}}}}
debug: [ELASTICSEARCH] loadByTerms > {"index":["stix_domain_entities_v2","stix_relations","stix_observables","work_jobs_index"],"_source_excludes":"rel_*","body":{"query":{"bool":{"must":[{"term":{"internal_id_key.keyword":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}},{"term":{"parent_types.keyword":"User"}}]}}}}
debug: [GRAKN - infer: false] updateAttribute - delete > match $x isa User, has internal_id_key "88ec0c6a-13ce-5e39-b486-354fe4a7084f", has password $del via $d; delete $d;
debug: [GRAKN - infer: false] updateAttribute - insert > match $x isa User, has internal_id_key "88ec0c6a-13ce-5e39-b486-354fe4a7084f"; insert $x has password "$2a$10$jTI3rzAXUMw1c11HUM2Va.4AgdGR9CA/2Y.bux8JR.1CZuW/oMdGW";
debug: [ELASTICSEARCH] loadByTerms > {"index":["stix_domain_entities_v2","stix_relations","stix_observables","work_jobs_index"],"_source_excludes":"rel_*","body":{"query":{"bool":{"must":[{"term":{"internal_id_key.keyword":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}},{"term":{"parent_types.keyword":"User"}}]}}}}
debug: [GRAKN - infer: false] updateAttribute - delete > match $x isa User, has internal_id_key "88ec0c6a-13ce-5e39-b486-354fe4a7084f", has updated_at $del via $d; delete $d;
debug: [GRAKN - infer: false] updateAttribute - insert > match $x isa User, has internal_id_key "88ec0c6a-13ce-5e39-b486-354fe4a7084f"; insert $x has updated_at 2020-03-04T13:40:32;
debug: [ELASTICSEARCH] loadByTerms > {"index":["stix_domain_entities_v2","stix_relations","stix_observables","work_jobs_index"],"_source_excludes":"rel_*","body":{"query":{"bool":{"must":[{"term":{"internal_id_key.keyword":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}},{"term":{"parent_types.keyword":"User"}}]}}}}
debug: [GRAKN - infer: false] updateAttribute - delete > match $x isa User, has internal_id_key "88ec0c6a-13ce-5e39-b486-354fe4a7084f", has modified $del via $d; delete $d;
debug: [GRAKN - infer: false] updateAttribute - insert > match $x isa User, has internal_id_key "88ec0c6a-13ce-5e39-b486-354fe4a7084f"; insert $x has modified 2020-03-04T13:40:32;
debug: [ELASTICSEARCH] loadByTerms > {"index":["stix_domain_entities_v2","stix_relations","stix_observables","work_jobs_index"],"_source_excludes":"rel_*","body":{"query":{"bool":{"must":[{"term":{"internal_id_key.keyword":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}},{"term":{"parent_types.keyword":"User"}}]}}}}
debug: [GRAKN - infer: false] Find > match $x isa Token;
    $rel(authorization:$x, client:$client) isa authorize;
    $client has internal_id_key "88ec0c6a-13ce-5e39-b486-354fe4a7084f"; get; offset 0; limit 1;
error: [OPENCTI] GraphQL initialization fail > TypeError: Cannot read property 'node' of undefined
    at /opt/opencti/build/index.js:1:61829
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async ae (/opt/opencti/build/index.js:1:68247)
    at async de (/opt/opencti/build/index.js:1:69456)
    at async I (/opt/opencti/build/index.js:1:101527)
    at async /opt/opencti/build/index.js:1:123139
    at async n.b (/opt/opencti/build/index.js:1:123042)
    at async /opt/opencti/build/index.js:1:841022
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by SamuelHassine** [comment]
Thanks everyone. I successfully reproduced the bug it will be fixed in the 3.0.3.

**Comment by toshiro92** [comment]
Ok @SamuelHassine I will test it once published then
